cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -1025,12 +1031,22 @@ private boolean isProgrammaticConfig(Configuration conf, String name) {
   @Nullable
   private Location createPluginArchive(Location targetDir) throws IOException {
     File pluginArchive = context.getPluginArchive();
-    if (pluginArchive == null) {
+    return copyFileToLocation(pluginArchive, targetDir);
+  }
+
+  /**
+   * Copies a plugin archive jar to the target location.
+   *
+   * @param targetDir directory where the archive jar should be created
+   * @return {@link Location} to the plugin archive or {@code null} if no plugin archive is available from the context.
+   */
+  private Location copyFileToLocation(File file, Location targetDir) throws IOException {
+    if (file == null) {
```

Comment:
Annotate the `file` parameter as `@Nullable` if null is allowed. Also update javadoc accordingly.